### PR TITLE
Add possibility to use your own M3GNet potential

### PR DESCRIPTION
**Pull request by QuantumChemist**
allow the possibility to use your own M3GNet potential, instead of the pretrained model only.

I want to run the `M3GNetRelaxMaker` and the `M3GNetStaticMaker` for the forcefield `PhononMaker`
https://github.com/materialsproject/atomate2/blob/06e4a715037ac1a86d7bfe3af5fb6b75236123bc/src/atomate2/forcefields/jobs.py#L258
with my own trained potential (passed in the `calculator_kwargs` to the PhononMaker). 
Therefore I added the possibility to use your own potential, instead of the pretrained model only.

### CODE REVIEW

**Comment by QuantumChemist** [comment]
Hi @utf , this PR is ready for review and merging. Thanks 😄

File: tests/forcefields/test_utils.py [hunk 1]
```diff
@@ -163,3 +163,27 @@ def test_fix_symmetry(fix_symmetry):
         assert symmetry_init["number"] == symmetry_final["number"] == 229
     else:
         assert symmetry_init["number"] != symmetry_final["number"] == 99
+
+
+def test_m3gnet_pot():
+    import matgl
+    from matgl.ext.ase import PESCalculator
+
+    m3gnet_calculator = ase_calculator(
```
Comment (QuantumChemist) [inline tests/forcefields/test_utils.py:172]: I actually want to implement it as 
```python
    m3gnet_calculator = ase_calculator(
        calculator_meta="MLFF.M3GNet",
        **{"path": "M3GNet-MP-2021.2.8-DIRECT-PES", "stress_weight": 2.0},
    )

    m3gnet_default = ase_calculator(  # uses "M3GNet-MP-2021.2.8-PES" per default
        calculator_meta="MLFF.M3GNet", **{"stress_weight": 2.0},
    )
```

and the test passes, but ruff-format changes it....

Comment (JaGeo) [reply tests/forcefields/test_utils.py:172]: Just define the two dicts

**Comment by JaGeo** [comment]
@QuantumChemist will merge once the tests pass

File: tests/forcefields/test_utils.py (hunk 1 above)
Comment (QuantumChemist) [reply tests/forcefields/test_utils.py]: yup. Just noticed it after pushing 🤣 so I didn't want you to merge it like that

**Comment by msehabibur** [comment]
I want to use my own matgl model to simulate phonon dos. I used this code but it did not work out. It shows relax_maker=ForceFieldRelaxMaker(ase_calculator=calculator, relax_cell=True)
TypeError: ForceFieldRelaxMaker.__init__() got an unexpected keyword argument 'ase_calculator'


`from pymatgen.core.structure import Structure
from atomate2.forcefields.flows.phonons import PhononMaker
from atomate2.forcefields.flows.elastic import ElasticMaker
from atomate2.forcefields.jobs import ForceFieldRelaxMaker
from jobflow import run_locally, SETTINGS
from pymatgen.phonon.bandstructure import PhononBandStructureSymmLine
from pymatgen.phonon.dos import PhononDos
from pymatgen.phonon.plotter import PhononBSPlotter, PhononDosPlotter
import matplotlib.pyplot as plt
import csv

# === Load your own MatGL model ===
from matgl import load_model
from matgl.ext.ase import PESCalculator

# Load your trained model from ./PES/
potential = load_model("./PES/")
calculator = PESCalculator(potential, stress_weight=2.0)

# === Load structure from CIF ===
structure = Structure.from_file("POSCAR.cif")

# === Create Phonon workflow using your model ===
phonon_flow = PhononMaker(
    min_length=15.0,
    store_force_constants=False,
    relax_maker=ForceFieldRelaxMaker(ase_calculator=calculator, relax_cell=True)
).make(structure=structure)

print("✅ Phonon workflow generated successfully.")

# === Optional: Elastic workflow using the same model ===
elastic_flow = ElasticMaker(
    bulk_relax_maker=ForceFieldRelaxMaker(ase_calculator=calculator, relax_cell=True, relax_kwargs={"fmax": 1e-6}),
    elastic_relax_maker=ForceFieldRelaxMaker(ase_calculator=calculator, relax_cell=False, relax_kwargs={"fmax": 1e-6}),
).make(structure=structure)

print("✅ Elastic workflow generated successfully.")

# === Run phonon job locally ===
run_locally(phonon_flow, create_folders=True)

# === Query phonon results ===
store = SETTINGS.JOB_STORE
store.connect()

result = store.query_one(
    {"name": "generate_frequencies_eigenvectors"},
    properties=["output.phonon_dos", "output.phonon_bandstructure"],
    load=True,
    sort={"completed_at": -1}
)

# === Load phonon objects ===
ph_bs = PhononBandStructureSymmLine.from_dict(result["output"]["phonon_bandstructure"])
ph_dos = PhononDos.from_dict(result["output"]["phonon_dos"])

# === Plot & save Phonon DOS ===
dos_plot = PhononDosPlotter()
dos_plot.add_dos("Phonon DOS", ph_dos)
fig_dos = dos_plot.get_plot().get_figure()
fig_dos.savefig("phonon_dos.png", dpi=300)
plt.close(fig_dos)

# === Plot & save Phonon Band Structure ===
bs_plot = PhononBSPlotter(ph_bs)
fig_bs = bs_plot.get_plot().get_figure()
fig_bs.savefig("phonon_bandstructure.png", dpi=300)
plt.close(fig_bs)

# === Export DOS CSV ===
with open("phonon_dos.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerow(["Frequency (THz)", "Total DOS"])
    writer.writerows(zip(ph_dos.frequencies, ph_dos.densities))

# === Export Band Structure CSV ===
with open("phonon_bandstructure.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerow(["Distance (1/Å)"] + [f"Branch {i+1} (THz)" for i in range(ph_bs.bands.shape[0])])
    for i, dist in enumerate(ph_bs.distance):
        writer.writerow([dist] + list(ph_bs.bands[:, i]))
    writer.writerow([])
    writer.writerow(["High Symmetry Points"])
    for label, dist in ph_bs.labels_dict.items():
        writer.writerow([label, dist])

print("✅ All phonon plots and CSV files saved.")
`

**Comment by msehabibur** [comment]
@QuantumChemist

**Comment by JaGeo** [comment]
@msehabibur 
Thank you for your interest in using atomate2. 
It is not possible to directly pass an ase calculator to the ForcefieldRelaxMakers as you do it. Instead, you need to write a new Maker based on the AseMaker class where you substitute the original calculator. Then, you should be able to use your own model.

**Pull request merged**